File: CS50/Problems/cash/cash.c
# include <cs50.h>
# include <stdio.h>
# include <math.h>

int main(void)
{
    //get change
    float euros;
    do
    {
        euros = get_float("change owed: ");
    }
    while (euros <= 0);
    
    // round change
    int change = round(euros * 100);
    
    // start coin counter
    int coins = 0;

    // loop through coins giving highest value 
    while (change > 0)
    {
        if (change >= 25)
        {
            change = change - 25;
            coins++;
        }
        else if (change >= 10)
        {
            change = change - 10;
            coins++;
        }
        else if (change >= 5)
        {
            change = change - 5;
            coins++;
        }
        else
        {
            change = change - 1;
            coins++;
        }
    }
    
    // display number of coins used
    printf("coins: %i\n", coins);
}    

File: CS50/Problems/sentimental-cash/cash.py
# import get float
from cs50 import get_float

# get change
money = 0
while money <= 0:
    money = get_float("Change owed: ")
    
# round change
m = money*100

# coin counter
c = 0

# loop to give change
while m > 0:
    if m >= 25:
        m -= 25
        c += 1
    elif m >= 10:
        m -= 10
        c += 1
    elif m >= 5:
        m -= 5
        c += 1
    else:
        m -= 1
        c += 1
       
# display num coins used       
print(c)